### PR TITLE
Add robert the bruce retries around mail sending

### PR DESCRIPTION
**Pull request by tank157**
I'm getting real tired of manual email resends, so this should help reduce that. I wasn't 100% sure which exception was being thrown, I think now based on the structure of the error it's probably a MailConnectException which inside it has a SocketConnectException brought on by an UnknownHostException, but I configured it to take either MailConnectException or UnknownHostException as a retry-able exception. It'll only retry 3 times just incase somehow the mail does get sent the first time, but I can't imagine how.

Sorry for all the `with-redefs` in the tests but not sure there was a better way around that.

[Here's a link to a recent email error so you can see the exception structure yourself](https://papertrailapp.com/groups/1634663/events?q=program%3Aemail-works%2Fc13549debd69&focus=805652436411891723).

### CODE REVIEW

**Comment by tank157** [comment]
@ddellacosta I wanted to get your take on this too, since it's now your project.

File: test/email_works/handlers_test.clj [hunk 1]
```diff
@@ -57,3 +58,40 @@
...
+
+(deftest smtp-send-test
+  (testing "recovers from a single UnknownHostException"
+    (let [call-count (atom 0)
+          send-fn (fn [_ _]
+                    (swap! call-count inc)
+                    (if (< 1 @call-count)
+                      :success
+                      (throw (java.net.UnknownHostException. "nope"))))]
+      (with-redefs [p/send-message send-fn]
```
Comment (ddellacosta) [inline test/email_works/handlers_test.clj]: Minor nitpick: I'd rather see `smtp-send` restructured as a multi-arity function taking the postal `send-message` function by default vs. the `with-redefs` here...but not a huge deal.

Comment (tank157) [inline test/email_works/handlers_test.clj]: It bothered me too, and I like your suggestion, it shall be done.

**Comment by tank157** [comment]
@ddellacosta How's this now?

**Comment by ddellacosta** [comment]
@tank157 Awesome!

**Pull request merged**